packages/sdk/src/wait_for_remote_peer.ts

```diff
@@ -35,42 +32,48 @@ export async function waitForRemotePeer(
   timeoutMs?: number
 ): Promise<void> {
   protocols = protocols ?? getEnabledProtocols(waku);
+  const connections = waku.libp2p.getConnections();
+
+  if (!waku.isStarted()) {
+    throw Error("Waku node is not started");
+  }
 
-  if (!waku.isStarted()) return Promise.reject("Waku node is not started");
+  if (connections.length > 0) {
+    const success = await waitForMetadata(waku.libp2p);
+
+    if (success) {
+      return;
```

Comment:
I don't understand the purpose of this return statement. Shouldn't there be some sort of check that the connected peer fulfills the protocol requirements before returning?

Reply:
that's actually straight to the point! 

I want to add this in the follow up PR, for now this is only movement of previous logic that was there

Reply:
here is follow up https://github.com/waku-org/js-waku/pull/2160